rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/RubricsServiceImpl.java

```diff
@@ -1522,6 +1523,181 @@ public Map<String, String> transferCopyEntities(String fromContext, String toCon
+
+                String description = criterion.getDescription();
+                if (StringUtils.isNotBlank(description)) {
+                    Element descriptionEl = doc.createElement("description");
+                    criterionEl.appendChild(descriptionEl);
+                    descriptionEl.appendChild(doc.createCDATASection(description));
+                }
+
+                Element ratingsEl = doc.createElement("ratings");
+                criterionEl.appendChild(ratingsEl);
+
+                criterion.getRatings().forEach(rating -> {
+
+                    Element ratingEl = doc.createElement("rating");
+                    ratingsEl.appendChild(ratingEl);
+                    ratingEl.setAttribute("title", rating.getTitle());
+                    ratingEl.setAttribute("points", Double.toString(rating.getPoints()));
+                    String ratingDescription = rating.getDescription();
+                    if (StringUtils.isNotBlank(ratingDescription)) {
+                        Element ratingDescriptionEl = doc.createElement("description");
+                        ratingEl.appendChild(ratingDescriptionEl);
+                        ratingDescriptionEl.appendChild(doc.createCDATASection(ratingDescription));
+                    }
+                });
+            });
+        });
+
+        stack.pop();
+
+        results.append("completed archiving ").append(getLabel()).append(" for site ").append(siteId).append(LINE_SEPARATOR);
+        return results.toString();
+    }
+
+    @Override
+    public String merge(String toSiteId, Element root, String archivePath, String fromSiteId, Map<String, String> attachmentNames, Map<String, String> userIdTrans, Set<String> userListAllowImport) {
+
+        StringBuilder results = new StringBuilder();
+        results.append("begin merging ").append(getLabel()).append(" for site ").append(toSiteId).append(LINE_SEPARATOR);
+
+        String currentUserId = sessionManager.getCurrentSessionUserId();
+
+        if (!root.getTagName().equals(getLabel())) {
+            log.warn("Tried to merge a non <{}> xml document", getLabel());
+            return "Invalid xml document";
+        }
+
+        NodeList rubricNodes = root.getElementsByTagName("rubric");
+
+        for (int i = 0; i < rubricNodes.getLength(); i++) {
+            Element rubricEl = (Element) rubricNodes.item(i);
+
+            String creatorId = currentUserId;
+
+            String originalCreatorId = rubricEl.getAttribute("creator");
+            if (StringUtils.isNotBlank(originalCreatorId)) {
+                try {
+                    userDirectoryService.getUser(originalCreatorId);
+                    creatorId = originalCreatorId;
+                } catch (UserNotDefinedException unde) {
+                }
+            }
+
+            RubricTransferBean rubricBean = new RubricTransferBean();
+            rubricBean.setOwnerId(toSiteId);
+            // TODO: Do we honour the original creator, or the current user?
```

Comment:
I vote for the creatorId - so far the only merge() that even looks at creatorId is Polls.  The problem of using the original is if you are moving server to server - the user Id from the archive is meaningless and a possibly a security hole if it is used in an Authz role.  I hope that the owner id is not an authz thing - and instead just some metadata.  If it is just metadata - creatorId is correct (i.e. for the traditional Site Archive Import in a batch jobs) with will be admin

Reply:
Owner id is the site where the rubric is hosted, ie: the siteId.

Reply:
The way the code is now the original creatorId is used if it is the user id of a valid Sakai user. If not, the current user, likely 'admin', is used.